### PR TITLE
fix: set do_request auth token precedence

### PR DESCRIPTION
**Pull request by hamidzr**
## Test Plan

ee tests run here https://github.com/determined-ai/determined-ee/pull/660
updated ver https://app.circleci.com/pipelines/github/determined-ai/determined-ee?branch=fix-oauthtest

## Commentary (optional)

we ran into an issue on ee after some test util changes. what was failing: https://app.circleci.com/pipelines/github/determined-ai/determined-ee/5110/workflows/27e580e4-a95d-4470-8859-0529c023ccbc/jobs/176006

@dzhu noticed that a header authorization token could be replaced by the cli auth singleton. This makes it clear what the order of the precedence is. 

we could try to preserve this method but it was becoming a bit awkward. open to suggestions.

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: e2e_tests/tests/api_utils.py [hunk 1]
```diff
@@ -24,11 +24,11 @@ def determined_test_session(
             credentials = authentication.Credentials("determined", "")
 
     murl = conf.make_master_url()
-    cert = certs.default_load(murl)
-    auth = authentication.Authentication(
+    certs.cli_cert = certs.default_load(murl)
```
Comment (hamidzr) [inline e2e_tests/tests/api_utils.py:27]: this diff is already in master

File: harness/determined/common/api/request.py [hunk 2]
```diff
@@ -108,23 +88,34 @@ def do_request(
...
+            h["Authorization"] = "Bearer {}".format(auth.get_session_token())
+        username = auth.get_session_user()
+    elif h.get("Authorization") is not None:
+        pass
+    elif authentication.cli_auth is not None:
+        if authenticated:
+            h["Authorization"] = "Bearer {}".format(authentication.cli_auth.get_session_token())
+        username = authentication.cli_auth.get_session_user()
+    elif h.get("Grpc-Metadata-x-allocation-token") is None:
+        allocation_token = authentication.get_allocation_token()
```
Comment (hamidzr) [inline harness/determined/common/api/request.py:112]: fixme: we only want to run this part `if authenticated`

File: harness/determined/common/api/request.py [hunk 3]
```diff
@@ -108,23 +88,34 @@ def do_request(
...
     if headers is None:
-        h = {}  # type: Dict[str, str]
+        h: Dict[str, str] = {}
     else:
         h = headers
 
+    if cert is None:
+        cert = certs.cli_cert
+
+    # set the token and username based on a perset precedence order.
```
Comment (rb-determined-ai) [inline harness/determined/common/api/request.py:99]: "perset"?  Should that be "preset"?

Maybe the comment should just say what the order is?

**Review by rb-determined-ai — APPROVED** [review]
It's still on my todo list to go through the master and harness code and [simplify our authentication token logic a bunch](https://determined-ai.slack.com/archives/C02PV33GSN5/p1670433561379169).

This change seems like a fine improvement in the mean time though.

Definitely do the fix you mentioned in your own comment.

**Pull request merged**